internal/cluster/validations/validations.go

```diff
@@ -300,7 +302,7 @@ func ValidateClusterUpdateVIPAddresses(ipV6Supported bool, cluster *common.Clust
 	targetConfiguration.APIVips = apiVips
 	targetConfiguration.IngressVips = ingressVips
 	targetConfiguration.UserManagedNetworking = params.UserManagedNetworking
-	targetConfiguration.HighAvailabilityMode = cluster.HighAvailabilityMode
+	targetConfiguration.ControlPlaneCount = cluster.ControlPlaneCount
```

Comment:
Control plane count currently can be updated, you need to add something like this before:

```
var ControlPlaneCount  = cluster.ControlPlaneCount
if params.ControlPlaneCount != nil {
      targetConfiguration.ControlPlaneCount = params.ControlPlaneCount
}
```